### ISSUE TITLE
Not able to use Edge webview WPF sample app when there are multiple RDP connections to the Server OS with the same user

### ISSUE THREAD

**Issue by RSN1907** [MicrosoftEdge/WebView2Feedback#3626]
**Description**
Not able to use Edge webview sample app when there are multiple simultaneous RDP connections to the Server OS with the same user ,the same is true for Microsoft Edge browser
 
![edgewebviewimage](https://github.com/MicrosoftEdge/WebView2Feedback/assets/42867049/021ba0b8-e155-4c5c-9c58-14eaca29c3ab)

**Version**
SDK:  LATEST SDK
Runtime:  Latest edge webview runtime
Framework:  WPF
OS: W2K16

**Regression**
Was this working before but has regressed? No
If yes, what version did this last work on? NA

**Repro Steps**

1) Create RDP connection to Server OS 2K16 via user1 and copy the edge webview2 WPF sample app to desktop folder
2) Open the sample app
3) While the session is active,create another RDP connection to the same Server OS machine with the same user user1
4) Try to open the sample app which was copied to desktop folder
 Expected:Edge webview sample app should open
Actual behaviour: Fails with error resource is already in use

**Screenshots**

**Additional context**

[AB#45786488](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/45786488)

**Comment by RSN1907** [comment]
@novac42 Can you kindly help on this issue?

**Comment by novac42** [comment]
Thanks for reporting the issue and sorry you are running into this. I've assigned this to a dev who can help follow up.

**Comment by RSN1907** [comment]
Thanks @novac42  for assigning

**Comment by plantree** [comment]
Hi @RSN1907,

Thanks for your feedbacks. Did you mean, Edge has the similar issue?

**Comment by RSN1907** [comment]
Yes correct

**Comment by plantree** [comment]
@RSN1907,

If Edge and WebView2 have the similar effect, it might be **by design**. Do you want Edge and WebView2 both support your feature request?

**Comment by RSN1907** [comment]
Yes

**Comment by RSN1907** [comment]
@plantree  If its new feature request ,can you kindly suggest if there is provision with Edge Webview such that it deletes userdatafolder by itself on closing

**Comment by plantree** [comment]
Hi @RSN1907,

Thanks for your feedbacks! If you want to submit a new feature request, could you summarize your whole needs, which will help us to better service you. Thanks.

**Comment by RSN1907** [comment]
@plantree  My requirements match with https://github.com/MicrosoftEdge/WebView2Feedback/issues/3637

**Comment by novac42** [comment]
@RSN1907 Could you please help us understand a bit more how this request matches #3637 ? My understanding is 3637 wants a way to not persist any user data between sessions. Do you also want this? May you help explain why you need to have multiple RDP connections with the same user?

**Comment by RSN1907** [comment]
@novac42  Our customers have the use case of having multiple RDP connections with the same user ,currently our application is failing for the second RDP session with the same user if the first session user is using the app , we use custom user data folder, since the folder is in use in one connection its failing in another. So we were thinking if msedgewebview2 by default can support this scenario by sharing UDF? If its not possible we can create UDF based on session ID to make it work but there is no clean way to delete user data folder on exit . So in order to solve this scenarios I was thinking if there is a way to have inmemory user data folder or atleast an option to configure such that User data folder is deleted on edge webview2 exit

**Comment by champnic** [comment]
@RSN1907 Thanks for the feedback. At this time we don't have plans to change the behavior of the user data folder to be in-memory only or automatically clean up. As you said, for this scenario please create separate UDFs for each connection. For cleaning up the UDF on exit, please use the [CoreWebView2Environment.BrowserProcessExited](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2environment.browserprocessexited?view=webview2-dotnet-1.0.2088.41) event to know when it is safe to delete the UDF.
